File: src/main/java/jp/itohiro/intellij/searchit/domain/QiitaItem.java
package jp.itohiro.intellij.searchit.domain;

import com.google.api.client.util.Key;

public class QiitaItem implements JsonItem{
    private static final String SITE = "Qiita";

    @Key
    private String title;

    @Key
    private String created_at;

    @Key
    private String url;

    @Key
    private String raw_body;

    @Key
    private String body;

    @Override
    public String getSite() {
        return SITE;
    }

    @Override
    public String getTitle() {
        return this.title;
    }

    @Override
    public String getCreateTime() {
        return this.created_at;
    }
}

File: src/main/kotlin/jp/itohiro/intellij/searchit/domain/QiitaItem.kt
package jp.itohiro.intellij.searchit.domain

import com.google.api.client.util.Key

public class QiitaItem : JsonItem {

    @Key
    private var title: String? = null

    @Key
    private var created_at: String? = null

    @Key
    private var url: String? = null

    @Key
    private var raw_body: String? = null

    @Key
    private var body: String? = null

    override fun getSite(): String {
        return SITE
    }

    override fun getTitle(): String? {
        return this.title
    }

    override fun getCreateTime(): String? {
        return this.created_at
    }

    companion object {
        private val SITE = "Qiita"
    }
}
